### PR TITLE
fix: update default tokens on unsupported networks

### PR DESCRIPTION
**Pull request by ianlapham**
- currentlty the widget will crash when `ExtendedEther` is use as a token input on a network that has no native currency 
- this issue is only exposed through network switching (as we disable token select on those networks)
- when switching from unsupported to supported, a race condition causes ExtendedEther to be used on a network where there is no native currency 

-  to avoid this we should only use "NATIVE" as a token default if we have instantiated the native currency to be used within `ExtendedEther`

### CODE REVIEW

File: src/lib/hooks/swap/useSyncTokenDefaults.ts [hunk 1]
```diff
@@ -26,7 +27,8 @@ function useDefaultToken(
     address = defaultAddress[chainId]
   }
   const token = useToken(address)
-  if (chainId && address === 'NATIVE') {
+  // Only use native currency if chain ID is in supported chains. ExtendedEther will error otherwise.
+  if (chainId && address === 'NATIVE' && chainId in SupportedChainId) {
```
Comment (zzmp) [inline src/lib/hooks/swap/useSyncTokenDefaults.ts:31]: use `useOnSupportedNetwork` instead of reimplementing this logic.

**Review by zzmp — CHANGES_REQUESTED** [review]
Re-use `useOnSupportedNetwork`.
There are two codepaths that use native currency in `useSyncTokenDefaults` - be sure to fix the latter, too (see L46).

**Comment by ianlapham** [comment]
> Re-use `useOnSupportedNetwork`. There are two codepaths that use native currency in `useSyncTokenDefaults` - be sure to fix the latter, too (see L46).

Would both be using the same hook here?

**Review by JFrankfurt — APPROVED** [review]

File: src/lib/hooks/swap/useSyncTokenDefaults.ts [hunk 2]
```diff
@@ -41,12 +47,13 @@ export default function useSyncTokenDefaults({
...
   const updateSwap = useUpdateAtom(swapAtom)
   const { chainId } = useActiveWeb3React()
+  const onSupportedNetwork = useOnSupportedNetwork()
   const nativeCurrency = useNativeCurrency()
   const defaultOutputToken = useDefaultToken(defaultOutputTokenAddress, chainId)
   const defaultInputToken =
     useDefaultToken(defaultInputTokenAddress, chainId) ??
     // Default the input token to the native currency if it is not the output token.
-    (defaultOutputToken === nativeCurrency ? nativeCurrency : undefined)
+    (defaultOutputToken === nativeCurrency && onSupportedNetwork ? nativeCurrency : undefined)
```
Comment (zzmp) [inline src/lib/hooks/swap/useSyncTokenDefaults.ts:56]: Looks like I had a bug here - it should be `defaultOutputToken !== nativeCurrency && onSupportedNetwork`.

Comment (ianlapham) [reply src/lib/hooks/swap/useSyncTokenDefaults.ts]: Yep updating

**Review by zzmp — APPROVED** [review]

**Pull request merged**